lib/Common/Memory/Recycler.cpp

```diff
@@ -4526,7 +4526,8 @@ Recycler::FinishConcurrent()
 
         const BOOL forceFinish = flags & CollectOverride_ForceFinish;
 
-        if (forceFinish || !IsConcurrentExecutingState())
+        if (forceFinish || !IsConcurrentExecutingState()
+            || flags == FinishConcurrentOnIdleAtRoot) // UI thread is idle, help background thread a bit
```

Comment:
Wouldn't this block the UI thread?

Reply:
Giving current thread is idle, and the work is just help clearing the pages swept during last one second, so it should not be very heavy. I think it would have low chance to cause UI hang, maybe keep this change and we can remove this if it's showing in hang stack some day?